### ISSUE TITLE
Visible batch effect on latent space and pseudotime

### ISSUE THREAD

**Issue by CanacR** [LiQian-XC/sctour#17]
Hello,

I integrated 3 different scRNA-seq datasets using Harmony after individual SCT normalization, and wanted to use scTour on the resulting integrated dataset (~40K cells).

I used the raw UMI counts as required, but I have a clear batch effect when I plot the latent space-derived UMAP and the pseudotime:

<img width="674" alt="Image" src="https://github.com/user-attachments/assets/5bbda9f1-7c35-42e1-9062-a09fa0c18e1d" />

Is there a way to take this dataset/batch information into account to avoid this bias?

Here is the code I used:

```
## Model training - mandatory step for scTour ##
sc.pp.calculate_qc_metrics(SeuratObj, percent_top=None, log1p=False, inplace=True)

## Select highly variable genes ##
sc.pp.highly_variable_genes(SeuratObj, flavor='seurat_v3', n_top_genes=1000, subset=True)

## Training ##
SeuratObj.X = SeuratObj.X.astype(np.float32)
tnode = sct.train.Trainer(SeuratObj, loss_mode='nb', alpha_recon_lec=0.5, alpha_recon_lode=0.5, percent=0.2)
tnode.train()

## Pseudotime ##
SeuratObj.obs['ptime'] = tnode.get_time()

## Latent Space ##
   # zs represents the latent z from variational inference, and pred_zs represents the latent z from ODE solver
   # mix_zs represents the weighted combination of the two, which is used for downstream analysis
mix_zs, zs, pred_zs = tnode.get_latentsp(alpha_z=0.5, alpha_predz=0.5)
SeuratObj.obsm['X_TNODE'] = mix_zs

   # UMAP embedding based on the inferred latent space
SeuratObj = SeuratObj[np.argsort(SeuratObj.obs['ptime'].values), :]
sc.pp.neighbors(SeuratObj, use_rep='X_TNODE', n_neighbors=15)
sc.tl.umap(SeuratObj, min_dist=0.1)

sc.pl.embedding(SeuratObj, basis="X_umap", color='ptime', show=False, frameon=False)
sc.pl.embedding(SeuratObj, basis="X_umap", color='Dataset', show=False, frameon=False)
```
Thank you.

**Comment by LiQian-XC** [comment]
Hi,

Thanks for sharing this. scTour does not explicitly account for batch information, as the continuous-in-time transformations by neural ODE can mitigate batch effects to some extent.

For your data, I suggest trying the following:
1. Specify `batch_key` when selecting highly variable genes using `sc.pp.highly_variable_genes`.
2. Adjust the training parameters by increasing `alpha_recon_lode` (e.g., 0.8) and decreasing `alpha_recon_lec` (e.g., 0.2) in `sct.train.Trainer`.

I hope this helps improve your results. Let me know if you have any further questions!

Best,
Qian
